usagov/vote-gov-drupal
# VOTE-1660: Update language migration

## Jira ticket (required)
[VOTE-1660](https://bixal-projects.atlassian.net/browse/VOTE-1660)

## Description (optional)
Update vote_migration module to include the content translation imports of Russian, Haitian Creole, Portuguese, and Arabic.

We also added an install file to the module to conditionally rewrite the migration source urls to support making curl requests to the site itself.

## Deployment and testing (required, if applicable)
### Pre-deploy steps
None

### Post-deploy steps
None

### Testing steps
_(The following process can now be done from the admin migration page, /admin/structure/migrate/manage/nodes__state_territory/migrations)_
1. Enable the vote_migrate module via Drush `drush en vote_migrate`. 
3. Confirm the current migration status using `drush ms`. You should see that there are Unprocessed imports for state_territory_translations.
-- Confirm that there aren't any errors displayed in red if using the command line indicating 403 errors. This will confirm that the new install file works as expected.
5. Run the migration using `drush mim --group=nodes__state_territory`.
6. Run `drush ms` again to confirm that there aren't any more Unprocessed imports.
7. Login to the website as an admin and go to the Content admin page to confirm that the translated pages appear.
8. Once the content has been confirmed, uninstall the module using `drush pmu vote_migrate` and confirm that the associated configurations have also been removed by visiting /admin/structure/migrate.
